rust_src/src/fns.rs

```diff
@@ -513,4 +515,36 @@ pub fn nconc(args: &mut [LispObject]) -> LispObject {
     val
 }
 
+/// Return a copy of a list, vector, string, char-table or record.
+/// The elements of a list, vector or record are not copied; they are
+/// shared with the original.
+/// If the original sequence is empty, this function may return
+/// the same empty object instead of its copy.
+#[lisp_fn]
+pub fn copy_sequence(arg: LispObject) -> LispObject {
+    unsafe {
+        if arg.is_nil() {
+            arg
+        } else if arg.is_record() {
+            Frecord(PVSIZE(arg), arg.force_vector().contents)
```

Comment:
Look at bindings.rs for the type of `contents`.

Reply:
`record` is a rust defined function. This can simply be:
```suggestion
            alloc::record(arg.force_vector().as_mut_slice())
```